File: sdk/examples/python/basic_usage.py
"""odyssey-sdk (Python) — a runnable walkthrough of every resource.

Prerequisites: a real `services/api` instance reachable at the URL below
(see `sdk/examples/README.md` for how to stand one up with sample data).

Run from the repo root:

    uv run python sdk/examples/python/basic_usage.py [base_url]
"""

from __future__ import annotations

import sys

from odyssey_sdk import OdysseyAPINotFoundError, OdysseySDK


def main() -> None:
    base_url = sys.argv[1] if len(sys.argv) > 1 else "http://127.0.0.1:8000"
    client = OdysseySDK(base_url)

    print("health:", client.health())

    journeys = client.journeys.list()
    print(f"{len(journeys)} journey(s)")
    for j in journeys[:3]:
        print(" -", j.journey_id)

    if journeys:
        detail = client.journeys.get(journeys[0].journey_id)
        print("first journey detail:", detail.journey_id, "complete =", detail.complete)

    try:
        client.journeys.get("does-not-exist")
    except OdysseyAPINotFoundError:
        print("journeys.get('does-not-exist') -> 404, as expected")

    print("datasets:", [d.name for d in client.datasets.list()])
    print("models:", [m.name for m in client.models.list()])
    print("eval runs:", [r.benchmark_name for r in client.runs.list()])
    print("exports:", [e.name for e in client.exports.list()])


if __name__ == "__main__":
    main()

File: sdk/examples/javascript/basic-usage.mjs
// @odyssey/sdk — a runnable walkthrough of every resource.
//
// Prerequisites: a real services/api instance reachable at the URL below
// (see sdk/examples/README.md), and @odyssey/sdk built (pnpm --filter
// @odyssey/sdk build) since this imports its dist/ output directly.
//
// Run from the repo root:
//
//   node sdk/examples/javascript/basic-usage.mjs [baseUrl]

import { OdysseySDK, OdysseyAPINotFoundError } from "../../javascript/dist/index.js";

async function main() {
  const baseUrl = process.argv[2] ?? "http://127.0.0.1:8000";
  const client = new OdysseySDK(baseUrl);

  console.log("health:", await client.health());

  const journeys = await client.journeys.list();
  console.log(`${journeys.length} journey(s)`);
  for (const j of journeys.slice(0, 3)) console.log(" -", j.journey_id);

  if (journeys.length > 0) {
    const detail = await client.journeys.get(journeys[0].journey_id);
    console.log("first journey detail:", detail.journey_id, "complete =", detail.complete);
  }

  try {
    await client.journeys.get("does-not-exist");
  } catch (err) {
    if (err instanceof OdysseyAPINotFoundError) {
      console.log("journeys.get('does-not-exist') -> 404, as expected");
    } else {
      throw err;
    }
  }

  console.log("datasets:", (await client.datasets.list()).map((d) => d.name));
  console.log("models:", (await client.models.list()).map((m) => m.name));
  console.log("eval runs:", (await client.runs.list()).map((r) => r.benchmark_name));
  console.log("exports:", (await client.exports.list()).map((e) => e.name));
}

main();
